wemake_python_styleguide/visitors/base.py

```diff
@@ -286,3 +286,60 @@ def run(self) -> None:
         for token in self.file_tokens:
             self.visit(token)
         self._post_visit()
+
+
+class BaseNodeTokenVisitor(ast.NodeVisitor, BaseVisitor):
+    """Allows storing violations during node tree traversal with real values."""
+
+    def __init__(
+        self,
+        options: ValidatedOptions,
+        tree: ast.AST,
+        file_tokens: Sequence[tokenize.TokenInfo],
+        **kwargs,
+    ) -> None:
+        """Creates new ``ast`` based instance with tokens."""
+        super().__init__(options, **kwargs)
+        self.tree = tree
+        self.file_tokens = file_tokens
+        self._token_index = -1
+        self._token_dict: dict[tuple[int, int], tokenize.TokenInfo] = {}
+
+    @final
+    @classmethod
+    def from_checker(
+        cls: type['BaseNodeTokenVisitor'], checker
+    ) -> 'BaseNodeTokenVisitor':
+        """Constructs visitor instance from the checker."""
+        return cls(
+            options=checker.options,
+            filename=checker.filename,
+            file_tokens=checker.file_tokens,
+            tree=checker.tree,
+        )
+
+    def visit(self, tree: ast.AST) -> None:
+        """
+        Visits a node.
+
+        Modified version of :class:`ast.NodeVisitor.visit` method.
+        We need this to modify how visitors route.
+
+        Why? Because python3.8 now uses ``visit_Constant`` instead of old
+        methods like ``visit_Num``, ``visit_Str``, ``visit_Bytes``, etc.
+
+        Some classes do redefine this method to catch all nodes. This is valid.
```

Comment:
Can you please instead link to the `BaseNodeVisitor` docs? We don't need to duplicate the reasoning here.